src/sedtrails/transport_converter/format_converter.py

```diff
@@ -1,80 +1,544 @@
+        (keys: 'x', 'y', 'magnitude', each with time as first dimension)
+    fractions: int
+        Number of sediment fractions
+    bed_load_sediment: Dict[str, np.ndarray]
+        Bed load sediment transport in kg/m/s 
+        (keys: 'x', 'y', 'magnitude', each with time as first dimension)
+    suspended_sediment: Dict[str, np.ndarray]
+        Suspended sediment transport in kg/m/s 
+        (keys: 'x', 'y', 'magnitude', each with time as first dimension)
+    water_depth: np.ndarray
+        Water depth in meters (with time as first dimension)
+    mean_bed_shear_stress: np.ndarray
+        Mean bed shear stress in pascal (with time as first dimension)
+    max_bed_shear_stress: np.ndarray
+        Max bed shear stress in pascal (with time as first dimension)
+    sediment_concentration: np.ndarray
+        Suspended sediment concentration in kg/m^3 (with time as first dimension)
+    nonlinear_wave_velocity: Dict[str, np.ndarray]
+        Nonlinear wave velocity in m/s 
+        (keys: 'x', 'y', 'magnitude', each with time as first dimension)
     """
     
-    time: int # current time in seconds
-    x: int # spatial coordinate in x direction
-    y: int # spatial coordinate in y direction
-    time_step: int # time step in seconds
-    bed_level: float # bed level in meters
-    depth_avg_flow_velocity: float # depth average flow velocity in m/s
-    fractions: int # number of fractions
-    bed_load_sediment: float # bed load sediment transport in kg/m/s
-    suspended_sediment: float # suspended sediment transport in kg/m/s
-    water_depth: float # water depth in meters
-    mean_bed_shear_stress: float # mean bed shear stress in pascal
-    max_bed_shear_stress: float # max bed shear stress in pascal
-    sediment_concentration: float # suspended sediment concentration in kg/m^3
-    nonlinear_wave_velocity: float # nonlinear wave velocity in m/s
-    previous = field(default=None, init=False)  # previous data in the sequence
-    next= field(default=None, init=False)  # next data in the sequence
-    
-    @property
-    def previous(self):
-        return self._previous
-    
-    @property
-    def next(self):
-        return self._next
-    
-    @previous.setter
-    def previous(self, data_sequence):
-        self._previous = data_sequence
-
-    @next.setter
-    def next(self, data_sequence):
-        self._next = data_sequence
-
-    def data_instance(self):
+    times: np.ndarray
+    reference_date: np.datetime64
+    x: np.ndarray
+    y: np.ndarray
+    bed_level: np.ndarray
+    depth_avg_flow_velocity: Dict[str, np.ndarray]
+    fractions: int
+    bed_load_sediment: Dict[str, np.ndarray]
+    suspended_sediment: Dict[str, np.ndarray]
+    water_depth: np.ndarray
+    mean_bed_shear_stress: np.ndarray
+    max_bed_shear_stress: np.ndarray
+    sediment_concentration: np.ndarray
+    nonlinear_wave_velocity: Dict[str, np.ndarray]
+    
+    def __getitem__(self, time_idx: int) -> Dict:
```

Comment:
this might sound very picky, but  'idx' is only 2 letter away of being more meaningful. Wouldn't be better to use the full word?